### PR TITLE
fix/4068 batchInsert correct definitions

### LINKED ISSUE #4068: batchInsert fix return type in typings
# Environment

Knex version: latest
Database + version: any
OS: any

Select applicable template from below.
If issue is about oracledb support, tag @ atiertant. For MSSql tag @ smorey2.
Rest of dialects don't need tags.
If issue is about TypeScript definitions, tag @lorefnon.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do
knex.batchInsert as for documentation returns knex.QueryBuilder
actually it is Promise & {returning, transacting}

### PR DESCRIPTION
**Pull request by maximelkin**
Closes #4068

### CODE REVIEW

**Comment by kibertoad** [comment]
@maximelkin One of the TSD tests are failing, try running `npm run lint:types` (or just `tsd`) locally.

File: test-tsd/batchInsert.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  // assert bad column in returning gives error
+  expectError(
+    knex.batchInsert('table', [{aa2: 'string', b: false, c: 12}], 12)
+      .returning('bad_column')
+  );
+
+  // assert any bad column from returning list gives error
+  // tsd gives error on this, lol
+  /*expectError(
```
Comment (kibertoad) [inline test-tsd/batchInsert.test-d.ts:38]: Do you think that tsd is wrong here? If there is an easy self-contained repro possible, perhaps an issue at https://github.com/SamVerschueren/tsd/issues could be opened?

**Comment by maximelkin** [comment]
tsc typecheck failing pipeline, I will try to fix (probably disable typecheck on them idk)

**Comment by maximelkin** [comment]
https://github.com/knex/knex/pull/4138 @kibertoad

File: test-tsd/batchInsert.test-d.ts (hunk 1 above)
Comment (maximelkin) [reply test-tsd/batchInsert.test-d.ts:38]: https://github.com/SamVerschueren/tsd/issues/87

Comment (kibertoad) [reply test-tsd/batchInsert.test-d.ts:38]: Thank you. Can you add the reference to that bug report next to the commented out code?

Comment (maximelkin) [reply test-tsd/batchInsert.test-d.ts:38]: I will try to fix it fast in tsd

Comment (maximelkin) [reply test-tsd/batchInsert.test-d.ts:38]: does tsd have any caches? I cant reproduce it anymore on extracted example

Comment (kibertoad) [reply test-tsd/batchInsert.test-d.ts:38]: Not tsd per se, but TypeScript definitely has some caches, so that might have been a factor.

Comment (maximelkin) [reply test-tsd/batchInsert.test-d.ts:38]: Oh, i copied stacktrace from wrong place

Comment (maximelkin) [reply test-tsd/batchInsert.test-d.ts]: Error pops up because this expression give 2 errors, but expectError allows only one

Comment (maximelkin) [reply test-tsd/batchInsert.test-d.ts]: alternative fix is expectError(expectError(<expression>))

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Thanks!

**Pull request merged**